### ISSUE TITLE
JWKS SSL opts in per-listener authentication not applied correctly in EMQX 5.7.1+

### ISSUE THREAD

**Issue by flbraun** [emqx/emqx#13586]
### What happened?

I'm running EMQX 5.7.0 and attempted to upgrade to 5.7.2 (the issue also exists with 5.7.1). I'm using the official Docker images.

Assume the following config:
```console
listeners.ssl {
  mylistener {
      enabled = true
      bind = "0.0.0.0:8885"
      ssl_options {
        cacertfile = "/certs/ca.fullchain.crt"
        certfile = "/certs/broker/broker.crt"
        keyfile = "/certs/broker/broker.key"
        verify = "verify_none"
        fail_if_no_peer_cert = false
        versions = ["tlsv1.3", "tlsv1.2"]
      }
      authentication = [
        {
          enable = true
          mechanism = "jwt"
          use_jwks = true
          endpoint = "https://my.jwks.endpoint"
          refresh_interval = 300
          from = "password"
        }
      ]
    }
}
```
In version 5.7.0, the listener authentication worked flawlessly. After upgrading to 5.7.1+ EMQX emits the following warning message on start, and fails to authenticate any connection on that listener:
```console
2024-08-07T11:22:26.385093+00:00 [warning]     msg: failed_to_request_jwks_endpoint
    reason: {failed_connect,
                [{to_address,{"my.jwks.endpoint",443}},
                 {inet,
                     [inet],
                     {options,incompatible,
                         [{verify,verify_peer},{cacerts,undefined}]}}]}
    endpoint: https://my.jwks.endpoint
```
If I interpret the log message correctly EMQX complains that it cannot use the SSL option `verify = verify_peer` when no cert authority is given. That sound logical, but according to the [5.7.1 documentation](https://docs.emqx.com/en/emqx/v5.7.1/hocon/#V-authentication-S-jwt_jwks-ssl-S-ssl_client_opts-verify) the default value of `verify` should still be `verify_none`!

Interestingly, explicitly setting `verify = verify_none` yields the exact same warning message.

### What did you expect to happen?

I expect the documented default and/or explicitly set value to be correctly applied.

### EMQX version

Docker image emqx/emqx:5.7.2 (sha256:6c0a4e7ad6bb4168d85ff24d93cc29e4394f908f80a6649bff60d2df98aeb0e3)
<details>

```console
$ ./bin/emqx_ctl broker
sysdescr  : EMQX
version   : 5.7.2
datetime  : 2024-08-07T11:40:33.650546416+00:00
uptime    : 5 minutes, 58 seconds
```

</details>

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here
```

</details>

### Log files

<details>

</details>

**Comment by qzhuyan** [comment]
if you still get 
```
                     {options,incompatible,
                         [{verify,verify_peer},{cacerts,undefined}]}}]}
```

then it is a bug,. `cacerts` should be omited instead of `{cacerts,undefined}`.

`cacerts` should be deprecated.

**Comment by qzhuyan** [comment]
may relates to #13257

@flbraun 
for 
> Interestingly, explicitly setting verify = verify_none yields the exact same warning message.

you set it via dashboard or via the config file?

**Comment by flbraun** [comment]
> you set it via dashboard or via the config file?

Config file.

**Comment by qzhuyan** [comment]
I managed to reproduce the issue. 

The issue can be solved by specifying the ` enable = true ` such like
```
...
      authentication = [
        {
          enable = true
          mechanism = "jwt"
          use_jwks = true
          endpoint = "https://my.jwks.endpoint"
          refresh_interval = 300
          from = "password"
          ssl.enable = true
        }
      ]
...
```

for record, the ssl options are not enabled in the config.
```
 emqx_config:get([listeners,ssl,mylistener]).
#{bind => {{0,0,0,0},8885},
  tcp_options =>
      #{buffer => 4096,high_watermark => 1048576,
        keepalive => "none",nodelay => true,reuseaddr => true,
        send_timeout => 15000,send_timeout_close => true,
        backlog => 1024,active_n => 100},
  authentication =>
      [#{ssl =>
             #{depth => 10,hibernate_after => 5000,verify => verify_none,
               enable => false,ciphers => [],log_level => notice,
               versions => ['tlsv1.3','tlsv1.2'],
               secure_renegotiate => true,reuse_sessions => true,
               user_lookup_fun => {fun emqx_tls_psk:lookup/3,undefined}},
         mechanism => jwt,pool_size => 8,enable => true,
         from => password,refresh_interval => 300,use_jwks => true,
         verify_claims => [],endpoint => "https://my.jwks.endpoint",
         disconnect_after_expire => true,
         acl_claim_name => <<"acl">>}],
  enable => true,
  ssl_options =>
      #{depth => 10,hibernate_after => 5000,verify => verify_none,
        ciphers => [],log_level => notice,
        versions => ['tlsv1.3','tlsv1.2'],
        client_renegotiation => true,fail_if_no_peer_cert => false,
        certfile => <<"/certs/broker/broker.crt">>,
        keyfile => <<"/certs/broker/broker.key">>,
        cacertfile => <<"/certs/ca.fullchain.crt">>,
        honor_cipher_order => true,secure_renegotiate => true,
        reuse_sessions => true,
        user_lookup_fun => {fun emqx_tls_psk:lookup/3,undefined},
        gc_after_handshake => false,handshake_timeout => 15000,
        ocsp =>
            #{refresh_interval => 300000,enable_ocsp_stapling => false,
              refresh_http_timeout => 15000},
        enable_crl_check => false},
  max_connections => infinity,
  access_rules => ["allow all"],
  acceptors => 16,proxy_protocol => false,
  proxy_protocol_timeout => 3000,zone => default,
  enable_authn => true,mountpoint => <<>>}
```

So OTP just use OTP defaults which is `verify_peer`

**Comment by flbraun** [comment]
Oh, that makes a lot of sense. The Docker images for 5.7.1+ are built on a newer OTP version, right?
I would have never thought about that, although I read about that change in the patch notes.

I successfully restored the old behavior by explicitly setting
```console
ssl {
  enable = true
  verify = "verify_none"
}
```
in the listener's authentication configuration.

Closing the ticket, thank you for your quick help!
